src/blockchains/eos/index.ts

```diff
@@ -36,8 +36,8 @@ export class EOS implements Blockchain {
         return [EOS.getShortestPath(sourcePath, targetPath)];
     }
 
-    async getRates(paths: Token[][], amount: string): Promise<string[]> {
-        return await Promise.all(paths.map(path => this.getRateByPath(path, amount)));
+    async getRates(paths: Token[][], amounts: string[]): Promise<string[][]> {
+        return Promise.all(amounts.map(amount => Promise.all(paths.map(path => this.getRateByPath(path, amount)))));
```

Comment:
There's an extra `Promise.all` here IMO.

Reply:
And for consistency, I'd keep the `await` (although it's not mandatory in a single-line function).

Reply:
Nice catch, thanks! I changed back to `return await`.

Re: extra `Promise.all`, we need two otherwise it resolves to `Promise<string[]>[]`. 

I changed it to two lines and added a comment to make it more clear. Let me know which you prefer :)

You can test in REPL by pasting the following:
```
    f = (x) => { return new Promise((resolve, reject) => resolve(x*2)) }
    testFunction = () => {
        const promises = Array(5).fill(['']).map(async (_, i) => {
            return await Promise.all([f(i), f(i*2)]);
        })
        return Promise.all(promises);
    }
    x = testFunction()
    x
    
```